R/utils.R

```diff
@@ -48,4 +48,107 @@ dip.file.exists <- function(x) {
   }
 }
 
+#' Comment out html in the file that contains images and
+#' urls that can't be accessed inside the SRE
+#'
+#' @param readme path to README file. Default `"README.md"`
+#'
+#' @return logical value if README was changed.
+#' @noRd
+prepare_readme_for_ocwa <- function(readme = "README.md") {
+  readme_txt <- readme_txt_orig <- readLines(readme)
 
+  badges_lines <- which(grepl("badges: (start|end)", readme_txt))
+
+  badges_to_comment <- seq(badges_lines[1] + 1, max(badges_lines) - 1)
+
+  # Only do if not already commented
+  badges_to_comment <- badges_to_comment[!grepl("<\\!--", readme_txt[badges_to_comment])]
+  if (length(badges_to_comment)) {
+    readme_txt[badges_to_comment] <- gsub("(.*)", "<!-- \\1 -->", readme_txt[badges_to_comment])
+  }
+
+  img_lines <- !grepl("(?=<!--).*<img", readme_txt, perl = TRUE)
+
+  if (length(img_lines)) {
+    readme_txt[img_lines] <- gsub("(.*)(<img.+?/>)(.*)", "\\1 <!-- \\2 --> \\3",
+                                  readme_txt[img_lines])
+  }
+
+  changed <- !isTRUE(all.equal(readme_txt, readme_txt_orig))
+  if (changed) writeLines(readme_txt, readme)
+  changed
+}
+
+#' Remove files from repo that are listed in `_ocwaignore` file.
+#'
+#' @return files that were removed
+#' @noRd
+process_ocwaignore <- function() {
+  file_globs <- readLines("_ocwaignore")
+  files <- Sys.glob(file_globs)
+  go <- utils::askYesNo(paste0("The following files will be deleted. Is that ok?\n",
+                  paste(files, collapse = "\n"), "\n"))
+  if (isTRUE(go)) {
+    unlink(files)
+    files
+  }
+}
+
+#' Prepare repo for import into the SRE through OCWA
+#'
+#' Removes files listed in `_ocwaignore` and processes `README.md`
+#' so that it is not using images or URLs that can't be accessed in the SRE.
+#'
+#' Then checks out a branch in `{dipr}` and commits those changes to that branch.
+#' Then use that branch to import into OCWA.
+#'
+#' @param branch branch name to prepare for OCWA. Default `"ocwa-import"`
+#'
+#' @return hash of commit
+#' @export
+ocwa_branch_export <- function(branch = "ocwa-import") {
+  if (!requireNamespace("gert", quietly = TRUE)) {
+    stop(
+      "The gert package is required for this function"
+    )
+  }
+
+  curr_branch <- gert::git_branch()
+
+  proceed <- utils::askYesNo(paste0("This will prepare a branch named ", branch,
+                      "\nwith the current changes in ", curr_branch,
+                      ".\nWould you like to proceed?\n"))
+  stopifnot(isTRUE(proceed))
```

Comment:
Maybe a more informative message?
```suggestion
  if(!isTRUE(proceed)) stop("export branch not created", call. = FALSE)
```